store/tikv/2pc.go

```diff
@@ -929,13 +929,15 @@ func (actionCommit) handleSingleBatch(c *twoPhaseCommitter, bo *Backoffer, batch
 		if c.mu.committed {
 			// No secondary key could be rolled back after it's primary key is committed.
 			// There must be a serious bug somewhere.
-			logutil.BgLogger().Error("2PC failed commit key after primary key committed",
+			logutil.Logger(bo.ctx).Error("2PC failed commit key after primary key committed",
 				zap.Error(err),
-				zap.Uint64("txnStartTS", c.startTS))
+				zap.Uint64("txnStartTS", c.startTS),
+				zap.Uint64("commitTS", c.commitTS),
+				zap.ByteStrings("keys", batch.keys))
```

Comment:
Will it print a very long message.

Reply:
Maybe, as this means a fatal error which rarely occurs, more information could do some help

Reply:
What will this output look like? Is it readable?

Reply:
It will be like this
```
[\"t\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0015_r\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0001\",\"t\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0015_i\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0001\\u0004\\u0005\\ufffd=\\u0005\\ufffd\\ufffd\\u0000\\u0000\\u0003\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0001\",\"t\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0015_i\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0001\\u0004\\u0005\\ufffd\\ufffd\\ufffd\\ufffd\\ufffd\\u0000\\u0000\\u0003\\ufffd\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0001\"]"]
```

Reply:
It is better to convert it to hex string.

Reply:
If so, how about using `prettyWriteKey`, and avoid the `keys` too much, should do a cut off?

Reply:
yes, `prettyWriteKey` is another choice, but it makes the log even longer. I think hex is enough and robust.